docker/ACL.yaml

```diff
@@ -0,0 +1,467 @@
+# This file defines the roles allowed for MBean operations
+#
+# The definition of ACLs for JMX operations works as follows:
+#
+# Based on the ObjectName of the JMX MBean, a key composed with the ObjectName
+# domain, followed by the 'type' attribute optionally, can be declared, using the
+# convention <domain>.<type>.
+# For example, the 'java.lang.Threading' key for the MBean with the following
+# objectName: java.lang:type=Threading can be declared. A more generic key with
+# the domain only can be declared (e.g. java.lang). A 'default' top-level key can
+# also be declared.
+# A key can either be an unordered or ordered map, whose keys can either be
+# string or regexp, and whose values can either be string or array of strings,
+# that represent roles that are allowed to invoke the MBean member.
+#
+# The system looks for required roles using the following process:
+#
+# The most specific key is tried first. E.g. for the
+# above example, the java.lang.Threading key is looked up first.
+# If the most specific key does not exist, the domain-only key is looked up,
+# otherwise, the 'default' key is looked up.
+# Using the matching key, the system looks up its map value for:
+#   1. An exact match for the operation invocation, using the operation
+#      signature, and the invocation arguments, e.g.:
+#      uninstall(java.lang.String)[0]: [] # no roles can perform this operation
+#   2. A regexp match for the operation invocation, using the operation
+#      signature, and the invocation arguments, e.g.:
+#      /update\(java\.lang\.String,java\.lang\.String\)\[[1-4]?[0-9],.*\]/: admin
+#      Note that, if the value is an ordered map, the iteration order is guaranteed,
+#      and the first matching regexp key is selected;
+#   3. An exact match for the operation invocation, using the operation
+#      signature, without the invocation arguments, e.g.:
+#      delete(java.lang.String): admin
+#   4. An exact match for the operation invocation, using the operation
+#      name, e.g.:
+#      dumpStatsAsXml: admin, viewer
+# If the key matches the operation invocation, it is used and the process will not
+# look for any other keys. So the most specific key always takes precedence.
+# Its value is used to match the role that impersonate the request, against the roles
+# that are allowed to invoke the operation.
+# If the current key does not match, the less specific key is looked up
+# and matched following the steps 1 to 4 above, up until the 'default' key.
+# Otherwise, the operation invocation is denied.
+
+default:
+  - list: admin, viewer
+  - read: admin, viewer
+  - search: admin, viewer
+  - /list.*/: admin, viewer
+  - /get.*/: admin, viewer
+  - /is.*/: admin, viewer
+  - /set.*/: admin
+  - /.*/: admin
```

Comment:
It's a bit non-intuitive to me that only the `default` requires a list but other keys use a map for the operation definitions. Is it a bad idea to use a map for both?

Reply:
The current rules declared in the `default` keys intersect, mostly because of the `/.*/: admin` rule. So the precedence of other rules has to be guaranteed, and using a list instead of a map provides that guarantee. I haven't used a list elsewhere, as there should be no key with intersecting rules.

Using a map is less verbose, and probably has better direct lookup performance, but the rules must not intersect. Otherwise using a list is an option.

Reply:
OK, that makes sense. So only the `default` section requires a ordered list. Probably it would be good to have a line of comment that explains the reasoning why it requires a ordered list?

Reply:
Right, I've just added a comment in eba473e2bd1e511bc8f180d1921917cbc4268d09.